cmake/Protobuf.cmake

```diff
@@ -13,30 +13,75 @@ macro(p4c_obtain_protobuf)
     endif()
     find_package(Protobuf CONFIG)
     if(NOT Protobuf_FOUND)
-        find_package(Protobuf REQUIRED)
+      find_package(Protobuf REQUIRED)
     endif()
+    # Backwards compatibility
+    # Define camel case versions of input variables
+    foreach(UPPER
+        PROTOBUF_SRC_ROOT_FOLDER
+        PROTOBUF_IMPORT_DIRS
+        PROTOBUF_DEBUG
+        PROTOBUF_LIBRARY
+        PROTOBUF_PROTOC_LIBRARY
+        PROTOBUF_INCLUDE_DIR
+        PROTOBUF_PROTOC_EXECUTABLE
+        PROTOBUF_LIBRARY_DEBUG
+        PROTOBUF_PROTOC_LIBRARY_DEBUG
+        PROTOBUF_LITE_LIBRARY
+        PROTOBUF_LITE_LIBRARY_DEBUG
+        )
+        if (DEFINED ${UPPER})
+            string(REPLACE "PROTOBUF_" "Protobuf_" Camel ${UPPER})
+            if (NOT DEFINED ${Camel})
+                set(${Camel} ${${UPPER}})
+            endif()
+        endif()
+    endforeach()
```

Comment:
Why is this needed? Looking at https://cmake.org/cmake/help/v3.6/module/FindProtobuf.html it seems the "new" versions are defined since CMake 3.6, we require at least 3.16 for p4c. Or is this coming from some other source? How did it work before?

Reply:
We are now using the CamelCase version of variables. It seemed prudent to add this conversion to be backwards compatible. But I happily remove it if we can assume that builds with users' existing systems.

Reply:
It seems the `FindPackge` module for Protobuf does define the "old-style" names itself (which is why the used until now). So I don't think we should not it ourselves.

And since we don't need to keep compatibility with CMake < 3.6 the worst thing that could happen is that someone will have to find this change and update their cmake when updating p4c.